addons/website/controllers/main.py

```diff
@@ -476,6 +476,84 @@ def get_current_currency(self, **kwargs):
     # Search Bar
     # --------------------------------------------------------------------------
 
+    def _shorten_around_match(self, text, keyword, width=80, placeholder="..."):
+        """
+        Return a shortened version of `text`, ensuring `keyword` stays visible
+        within the returned text. If truncation is necessary, the output is
+        centered around the match and uses `placeholder` to indicate omitted
+        content.
+
+        Handles multi-word keywords by finding the first token match.
+
+            >>> _shorten_around_match("aaa bbb ccc ddd", "ccc", 12)
+            "...bb ccc..."
+
+        :param text: Text to shorten
+        :param keyword: Substring or multi-word phrase that must remain visible
+        :param width: Maximum length of the returned text
+        :param placeholder: String used to indicate omitted content
+
+        :return: Shortened text ensuring `keyword` remains visible if possible
+        """
+        if not text or not keyword or len(text) <= width:
+            return text
+
+        lower_text = text.lower()
+        lower_keyword = keyword.lower()
+
+        # Try exact match first
+        match_pos = lower_text.find(lower_keyword)
+
+        # If no exact match, break keyword into tokens and find first match
+        if match_pos == -1:
+            tokens = lower_keyword.split()
+            if tokens:
+                for token in tokens:
+                    match_pos = lower_text.find(token)
+                    if match_pos != -1:
+                        keyword = text[match_pos:match_pos + len(token)]
+                        break
```

Comment:
```suggestion
        if match_pos == -1:
            tokens = lower_keyword.split()
            for token in tokens:
                match_pos = lower_text.find(token)
                if match_pos != -1:
                    keyword = text[match_pos:match_pos + len(token)]
                    break
```

No need of extra check here I guess. split() always return `[]` in case of blank string. Also, why do we need keyword to be re-assigned here?